app/assets/javascripts/discourse/app/components/user-menu/menu.js

```diff
@@ -0,0 +1,57 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { cached, tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+
+const DEFAULT_TAB_ID = "all-notifications";
+const DEFAULT_PANEL_COMPONENT = "user-menu/notifications-list";
+
+export default class UserMenu extends GlimmerComponent {
+  @tracked currentTabId = DEFAULT_TAB_ID;
+  @tracked currentPanelComponent = DEFAULT_PANEL_COMPONENT;
+
+  @cached
```

Comment:
I'm wondering if we need this.

> Ember uses caching based on trackable values to avoid updating large portions of the application.

Based on my understanding of the Ember docs above, `@cached` is only useful on getters when the function relies on a tracked value. The function below does not in this case.

Reply:
I added this because when this component is rendered, `topTabs` is called twice: once in the handlebars template and once in the `bottomTabs` property. Without caching, this property would be computed twice and I wanted to avoid this because in the original PR this property will call callbacks registered by plugins and I didn't want to invoke plugins' callbacks more than once when rendering the menu.

Reply:
I'm not sure if `@cache` is the right construct to use here since there is nothing glimmer related in the function we're "caching". See https://github.com/emberjs/rfcs/pull/566/files#diff-0b8d02be6975607dc087297205f31faa014af890bae5a73837ac0d8503abf573 on what `@cached` is meant to be use for. 

If `topTabs` is meant to be calculated once, we can just calculate it once in the `constructor` and assign it to an instance property of the class. Sort of like memoizing in Ruby.